ORST/Assets/Scripts/Core/UI/FollowCamera.cs

```diff
@@ -0,0 +1,137 @@
+using DG.Tweening;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace ORST.Core.UI {
+    public sealed class FollowCamera : MonoBehaviour {
+        private const float Y_TRIGGER_DISTANCE = 20;
+        private const float X_TRIGGER_DISTANCE = 20;
+
+        /// <summary>
+        ///     Should the component look at camera?
+        /// </summary>
+        [SerializeField] private bool lookAtCamera = true;
+
+        /// <summary>
+        ///     Should the component follow the camera?
+        /// </summary>
+        [SerializeField] private bool shouldFollowCamera = true;
+
+        /// <summary>
+        ///     Is the component following?
+        /// </summary>
+        [SerializeField] private bool isFollowing;
+
+        [SerializeField] private Vector3 positionOffeset = new Vector3(0, 0, 0);
+
+        /// <summary>
+        ///     The follow latency.
+        /// </summary>
+        [SerializeField] private float followLatency = 0.5f;
+
+        /// <summary>
+        ///     The distance from camera that the component should pe placed.
+        /// </summary>
+        [SerializeField] private float distanceFromCamera = float.NegativeInfinity;
+
+        /// <summary>
+        ///     The follow axis.
+        /// </summary>
+        [SerializeField] private AxisConstraint followAxis;
```

Comment:
I don't think we need documentation for private fields
```suggestion
        [SerializeField] private bool lookAtCamera = true;
        [SerializeField] private bool shouldFollowCamera = true;
        [SerializeField] private bool isFollowing;
        [SerializeField] private Vector3 positionOffeset = new Vector3(0, 0, 0);
        [SerializeField] private float followLatency = 0.5f;
        [SerializeField] private float distanceFromCamera = float.NegativeInfinity;
        [SerializeField] private AxisConstraint followAxis;
```